src/data/list/basic.lean

```diff
@@ -3009,8 +3009,62 @@ by convert filter_eq_nil.2 (λ _ _, id)
       take_while_append_drop l]
     else by rw [take_while, drop_while, if_neg pa, if_neg pa, nil_append]
 
+@[simp] theorem filter_leq_length (l : list α) : (filter p l).length ≤ l.length :=
+begin
+  induction l;
+  rw list.filter;
+  simp,
+  cases classical.em (p l_hd);
+  simp [h],
+  {
+    exact l_ih,
+  },
+  {
+    exact le_add_right l_ih
+  }
+end
+
+@[simp] theorem mem_filter_less_length {a : α} (l : list α) : a ∈ l → ¬p a → (filter p l).length < l.length :=
```

Comment:
You should have a read of the mathlib naming guide. This would be more like
```suggestion
@[simp] theorem length_filter_lt_of_mem {a : α} (l : list α) : a ∈ l → ¬p a → (filter p l).length < l.length :=
```

Reply:
Huh, should have guessed that would exist somewhere.
All of the names should be fixed now.